app/Exceptions/AuthenticationException.php

```diff
@@ -0,0 +1,13 @@
+<?php
+
+namespace App\Exceptions;
+
+use Exception;
+use Symfony\Component\HttpFoundation\Response;
+
+class AuthenticationException extends Exception
+{
+    public static function EmailOrPasswordAreNotValid(): self{
```

Comment:
Nome de métodos deve ser feito em _camelCase_

```suggestion
    public static function emailOrPasswordAreNotValid(): self{
```